gradle/publish-androidauto.gradle

```diff
@@ -0,0 +1,102 @@
+apply plugin: 'maven-publish'
+apply plugin: 'com.mapbox.sdkRegistry'
+apply from: file("${rootDir}/gradle/artifact-settings.gradle")
+apply from: "${rootDir}/gradle/kdoc-settings.gradle"
+
+def getVersionName() {
+    def cmd = ["git", "describe", "--tag", "--match", "androidauto-v*", "--abbrev=0"]
```

Comment:
Yeah I'll look into this. I was also looking to do something that would not block the core release flow if there is a bug in this one. But I can also look into making this script take the `androidauto-` part as a parameter.

Reply:
There are other parts that require a full copy paste of publish.gradle. The `sdkNameMap` is not a parameter 🤔

Reply:
Ok this is updated. I've left this as a separate commit for now https://github.com/mapbox/mapbox-navigation-android/pull/5773/commits/92bc88657c4b0f480dedca98e9e080e0424c7aba. That commit updates the main release pipeline

## Main release pipeline

publishing a snapshot for any module. current version is 2.5.0 so it bumps it to 2.6.0-SNAPSHOT
```
➜  mapbox-navigation-android git:(km-enable-android-auto-publisher) ./gradlew libnavui-speedlimit:publishToMavenLocal -Psnapshot=true

...
> Task :libnavui-speedlimit:publishReleasePublicationToMavenLocal
Multiple publications with coordinates 'com.mapbox.navigation:ui-speedlimit:2.6.0-SNAPSHOT' are published to repository 'mavenLocal'. The publications will overwrite each other!
```

publishing a release for any module. this will publish the last tag. in this case, 2.5.0-beta.3
```
➜  mapbox-navigation-android git:(km-enable-android-auto-publisher) ./gradlew libnavui-speedlimit:publishToMavenLocal

> Task :libnavui-speedlimit:publishReleasePublicationToMavenLocal
Multiple publications with coordinates 'com.mapbox.navigation:ui-speedlimit:2.5.0-beta.3' are published to repository 'mavenLocal'. The publications will overwrite each other!
```

## AndroidAuto release pipeline

Android Auto has defined a "RELEASE_TAG_PREFIX" in its gradle.properties. So it has its own release versions which defaults to 0.1.0 since none have been published

```
➜  mapbox-navigation-android git:(km-enable-android-auto-publisher) ./gradlew libnavui-androidauto:publishToMavenLocal -Psnapshot=true

> Task :libnavui-androidauto:publishReleasePublicationToMavenLocal
Multiple publications with coordinates 'com.mapbox.navigation:ui-androidauto:0.1.0-SNAPSHOT' are published to repository 'mavenLocal'. The publications will overwrite each other!
```

Takes the last release with a `android-v*` prefix (in this case it defaults to 0.1.0 since there hasn't been one published)
```
➜  mapbox-navigation-android git:(km-enable-android-auto-publisher) ./gradlew libnavui-androidauto:publishToMavenLocal

> Task :libnavui-androidauto:publishReleasePublicationToMavenLocal
Multiple publications with coordinates 'com.mapbox.navigation:ui-androidauto:0.1.0' are published to repository 'mavenLocal'. The publications will overwrite each other!
```